mne/viz/montage.py

```diff
@@ -27,47 +25,25 @@ def plot_montage(montage, scale_factor=20, show_names=True, kind='topomap',
     fig : instance of matplotlib.figure.Figure
         The figure object.
     """
-    from scipy.spatial.distance import cdist
     from ..channels import Montage, DigMontage
     from .. import create_info
 
-    if isinstance(montage, Montage):
-        ch_names = montage.ch_names
-        title = montage.kind
-    elif isinstance(montage, DigMontage):
-        ch_names = montage.point_names
-        title = None
-    else:
-        raise TypeError("montage must be an instance of "
-                        "mne.channels.montage.Montage or"
-                        "mne.channels.montage.DigMontage")
+    _validate_type(montage, types=(Montage, DigMontage), item_name='montage')
     _check_option('kind', kind, ['topomap', '3d'])
 
-    if isinstance(montage, Montage):  # check for duplicate labels
-        dists = cdist(montage.pos, montage.pos)
-        # only consider upper triangular part by setting the rest to np.nan
-        dists[np.tril_indices(dists.shape[0])] = np.nan
-        dupes = np.argwhere(np.isclose(dists, 0))
-        if dupes.any():
-            montage = deepcopy(montage)
-            n_chans = montage.pos.shape[0]
-            n_dupes = dupes.shape[0]
-            idx = np.setdiff1d(montage.selection, dupes[:, 1]).tolist()
-            logger.info("{} duplicate electrode labels found:".format(n_dupes))
-            logger.info(", ".join([ch_names[d[0]] + "/" + ch_names[d[1]]
-                                   for d in dupes]))
-            logger.info("Plotting {} unique labels.".format(n_chans - n_dupes))
-            montage.ch_names = [montage.ch_names[i] for i in idx]
-            ch_names = montage.ch_names
-            montage.pos = montage.pos[idx, :]
-            montage.selection = np.arange(n_chans - n_dupes)
```

Comment:
this was here for a reason. cc @larsoner 

+1 for unify attributes of montage and digmontage. point_names is not great. I also think that only channels have names.

Reply:
Indeed, `git blame` points to #4592 and #4403